diff --git a/CMakeLists.txt b/CMakeLists.txt
index 96130a4..b4f728c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,10 +51,12 @@ set_package_properties(LibIntl PROPERTIES TYPE REQUIRED
     PURPOSE "Needed for building KI18n unless glibc is the system libc implementation"
 )
 
-find_package(IsoCodes)
-set_package_properties(IsoCodes PROPERTIES TYPE RUNTIME
-    PURPOSE "Translation catalogs for countries, country subdivisions, languages and currencies"
-)
+if(NOT MSVC)
+    find_package(IsoCodes)
+    set_package_properties(IsoCodes PROPERTIES TYPE RUNTIME
+        PURPOSE "Translation catalogs for countries, country subdivisions, languages and currencies"
+    )
+endif()
 
 # KF6I18nMacros.cmake only needs to know the python executable path.
 # Due to CMake caching the variables, and KF6I18nMacros being included by KF6I18nConfig,
diff --git a/src/i18n/klocalizedcontext.h b/src/i18n/klocalizedcontext.h
index e895bc3..083f5e7 100644
--- a/src/i18n/klocalizedcontext.h
+++ b/src/i18n/klocalizedcontext.h
@@ -42,7 +42,7 @@
  * Use KLocalizedQmlContext or KLocalization::setupLocalizedContext
  * instead.
  */
-KI18N_DEPRECATED_VERSION(6, 8, "use KLocalizedQmlContext or KLocalization::setupLocalizedContext() from KF6::I18nQml instead")
+//KI18N_DEPRECATED_VERSION(6, 8, "use KLocalizedQmlContext or KLocalization::setupLocalizedContext() from KF6::I18nQml instead")
 class KI18N_EXPORT KLocalizedContext : public QObject
 {
     Q_OBJECT
